src/cpp/rtps/network/NetmaskFilterUtils.cpp

```diff
@@ -0,0 +1,186 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file NetmaskFilterUtils.cpp
+ */
+
+#include <algorithm>
+#include <cstdint>
+#include <sstream>
+#include <stdexcept>
+#include <unordered_map>
+
+#include <fastdds/rtps/attributes/ExternalLocators.hpp>
+#include <fastdds/rtps/transport/NetmaskFilterKind.hpp>
+
+#include <rtps/network/NetmaskFilterUtils.hpp>
+
+namespace eprosima {
+namespace fastdds {
+namespace rtps {
+namespace NetmaskFilterUtils {
+
+NetmaskFilterKind string_to_netmask_filter_kind(
+        const std::string& netmask_filter_str)
+{
+    static const std::unordered_map<std::string, NetmaskFilterKind> conversion_map =
+    {
+        {"OFF", NetmaskFilterKind::OFF},
+        {"AUTO", NetmaskFilterKind::AUTO},
+        {"ON", NetmaskFilterKind::ON}
+    };
```

Comment:
Same as before, can't because it's a map.
Regarding whether to group or not, I think it's better to have `fastdds::rtps::network_utils::string_to_netmask_filter_kind` rather than `fastdds::rtps::string_to_netmask_filter_kind`.